### PR TITLE
Add docs for `loguru` integration

### PR DESCRIPTION
**Pull request by PerchunPak**
(Again) mostly copied from `logging` docs.

My PR about adding `loguru` integration - https://github.com/getsentry/sentry-python/pull/1994.

## Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

**Comment by shanamatthews** [comment]
Thanks for writing docs! I'll wait to see what happens with the SDK PR before we review :)

**Comment by shanamatthews** [comment]
Commenting to keep open

**Comment by antonpirker** [comment]
Hey @PerchunPak !

Thanks for the docs (and sorry I did not see them earlier).
Could you please move the docs from the "guides" folder to this directory: https://github.com/getsentry/sentry-docs/blob/master/src/platforms/python/common/configuration/integrations/redis.mdx#L19

(Because the guides are kind of confusing, because all the pages of the documentation are duplicated for all the guides. I kind of want to go away from the guides and have the integrations documented in the configuration section. And I want to also have a better "index" of our integrations, because we have a lot and do not present them well)

Thanks!

**Comment by antonpirker** [comment]
Hey @PerchunPak !

I have changed some things around in the page to make the structure of the page more like the other integrations docs pages. 

But I am not allowed to push to your branch. Can you please give me access? (there is somewhere a checkbox to allow uptream persons to push)

**Comment by antonpirker** [comment]
Ignore my last comment @PerchunPak . Managed to push to the right branch.

**Comment by antonpirker** [comment]
Hey @shanamatthews ,
could you please have a look if this is proper English?

File: src/platforms/python/common/configuration/integrations/loguru.mdx [hunk 1]
```diff
@@ -0,0 +1,108 @@
+---
+title: Loguru
+description: Learn about using Sentry with loguru.
```
Comment (shanamatthews) [inline src/platforms/python/common/configuration/integrations/loguru.mdx:3]:
```suggestion
description: Learn about using Sentry with Loguru.
```

File: src/platforms/python/common/configuration/integrations/loguru.mdx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+sidebar_order: 80
+---
+
+The [Loguru](https://github.com/Delgan/loguru#readme) integration lets you capture log messages and send them to Sentry.
+
+Most of the work is handled by the [`logging`](/platforms/python/guides/logging) integration, so most of the examples there will work with loguru.
```
Comment (shanamatthews) [inline src/platforms/python/common/configuration/integrations/loguru.mdx:9]:
```suggestion
The Python [Logging](/platforms/python/guides/logging) integration provides most of the Loguru functionality and most examples on that page work with Loguru.
```

File: src/platforms/python/common/configuration/integrations/loguru.mdx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+## Install
+
+Install `sentry-sdk` from PyPI with the `loguru` extra.
+
+```bash
+pip install --upgrade 'sentry-sdk[loguru]'
+```
+
+## Configure
```
Comment (shanamatthews) [inline src/platforms/python/common/configuration/integrations/loguru.mdx:19]:
```suggestion
## Configure

Add `LoguruIntegration()` to your integrations list:
```

File: src/platforms/python/common/configuration/integrations/loguru.mdx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    dsn="___PUBLIC_DSN___",
+    integrations=[
+        LoguruIntegration(),
+    ],
+)
+```
+
+## Behavior
+
+By default log records with a level of `INFO` or higher will be added as breadcrumbs to Sentry events. For log records with a level of `ERROR` or higher a Sentry issue will be created:
```
Comment (shanamatthews) [inline src/platforms/python/common/configuration/integrations/loguru.mdx:35]:
```suggestion
By default, logs with a level of `INFO` or higher will be added as breadcrumbs to Sentry events. Sentry issue will be created for logs with a level of `ERROR` or higher: 
```

File: src/platforms/python/common/configuration/integrations/loguru.mdx [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+logger.info("I am a breadcrumb")
+logger.error("I am an event", extra=dict(bar=43))
+logger.exception("An exception happened")
+```
+
+- There will be an error event with the message `"I am an event"`.
+- `"I am a breadcrumb"` will be attached as a breadcrumb to that event.
+- `bar` will end up in the `extra` attributes of that event.
+- `"An exception happened"` will send the current exception from `sys.exc_info()` with the stack trace to the Sentry. If there's no exception, the current stack will be attached.
+- The debug message `"I am ignored"` will not be captured by Sentry. To capture it, you need to lower `level` to `DEBUG` in `LoguruIntegration`.
```
Comment (shanamatthews) [inline src/platforms/python/common/configuration/integrations/loguru.mdx:50]:
```suggestion
- An error event with the message `"I am an event"` will be created.
- `"I am a breadcrumb"` will be attached as a breadcrumb to that event.
- `bar` will end up in the `extra` attributes of that event.
- `"An exception happened"` will send the current exception from `sys.exc_info()` with the stack trace to Sentry. If there's no exception, the current stack will be attached.
- The debug message `"I am ignored"` will not be captured by Sentry. To capture it,  set `level` to `DEBUG` or lower in `LoguruIntegration`.
```

File: src/platforms/python/common/configuration/integrations/loguru.mdx [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+### Ignoring a logger
+
+Sometimes a logger can be noisy. You can ignore that logger by calling `ignore_logger`:
```
Comment (shanamatthews) [inline src/platforms/python/common/configuration/integrations/loguru.mdx:54]:
```suggestion
Loggers can be noisy. You can ignore a logger by calling `ignore_logger`:
```

File: src/platforms/python/common/configuration/integrations/loguru.mdx [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+
+```python
+# As most of the logic is proxied to `logging` integration, we use it instead of `loguru` integration.
```
Comment (shanamatthews) [inline src/platforms/python/common/configuration/integrations/loguru.mdx:57]:
```suggestion
# Since most of the logic is proxied to `logging` integration, we use it instead of the `loguru` integration.
```

Comment (shanamatthews) [inline src/platforms/python/common/configuration/integrations/loguru.mdx:57]:
```suggestion
# Since most of the logic is proxied to `logging` integration, we use it instead of the `loguru` integration.
```

File: src/platforms/python/common/configuration/integrations/loguru.mdx [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+- `"I am a breadcrumb"` will be attached as a breadcrumb to that event.
+- `bar` will end up in the `extra` attributes of that event.
+- `"An exception happened"` will send the current exception from `sys.exc_info()` with the stack trace to the Sentry. If there's no exception, the current stack will be attached.
+- The debug message `"I am ignored"` will not be captured by Sentry. To capture it, you need to lower `level` to `DEBUG` in `LoguruIntegration`.
+
+### Ignoring a logger
+
+Sometimes a logger can be noisy. You can ignore that logger by calling `ignore_logger`:
+
+```python
```
Comment (shanamatthews) [inline src/platforms/python/common/configuration/integrations/loguru.mdx:56]: This code snippet didn't make sense to me. At first I thought it was showing two ways of doing the same thing (one with `logging` and one with `loguru`, but I don't think that's the case.

Is it showing code that should be in two different files? If so, this should be in 2 separate code snippet boxes.

Maybe this would make sense to loguru users, and if so, feel free to ignore!

File: src/platforms/python/common/configuration/integrations/loguru.mdx [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+    dsn="___PUBLIC_DSN___",
+    integrations=[
+        sentry_loguru,
+    ],
+)
+```
+
+- `level`
+
+  The Sentry Python SDK will record log records with a level higher than or equal to `level` as breadcrumbs. Inversely, the SDK completely ignores any log record with a level lower than this one. If set to `None`, the SDK won't send log records as breadcrumbs.
```
Comment (shanamatthews) [inline src/platforms/python/common/configuration/integrations/loguru.mdx:95]:
```suggestion
  The Sentry Python SDK will record log records with a level higher than or equal to `level` as breadcrumbs. Inversely, the SDK ignores any log record with a level lower than this one. If set to `None`, the SDK won't send log records as breadcrumbs.
```

**Review by shanamatthews — APPROVED** [review]
This looks great! Really just one spot where I was confused on one of the code snippets. Thanks @PerchunPak for writing these up and @antonpirker for helping get them in great shape!

File: src/platforms/python/common/configuration/integrations/loguru.mdx (hunk 2 above)
Comment (PerchunPak) [reply src/platforms/python/common/configuration/integrations/loguru.mdx:9]: While correction `loguru -> Loguru` is right, the `logging -> Logging` is not. Because official Loguru docs mention its name with first capital character and `logging` docs mention itself as just logging (without capital and \`). I think it's quite confusing to use the word "logging" and library `logging` in one style, so isn't it better to mention `logging` with `?

File: src/platforms/python/common/configuration/integrations/loguru.mdx (hunk 7 above)
Comment (PerchunPak) [reply src/platforms/python/common/configuration/integrations/loguru.mdx:57]: Duplicate of https://github.com/getsentry/sentry-docs/pull/6617#discussion_r1190423916.

File: src/platforms/python/common/configuration/integrations/loguru.mdx [hunk 10]
```diff
@@ -51,50 +46,70 @@ logger.exception("An exception happened")
...
 ```python
 # As most of the logic is proxied to `logging` integration, we use it instead of `loguru` integration.
 from sentry_sdk.integrations.logging import ignore_logger
 
-
 ignore_logger("a.spammy.logger")
 
-# this will work with `logging`'s logger too
-logger = logging.getLogger("a.spammy.logger")
-logger.error("hi")  # no error sent to sentry
```
Comment (PerchunPak) [inline src/platforms/python/common/configuration/integrations/loguru.mdx]: Why did you remove these strings?

File: src/platforms/python/common/configuration/integrations/loguru.mdx (hunk 8 above)
Comment (PerchunPak) [reply src/platforms/python/common/configuration/integrations/loguru.mdx:56]: Yes, the example is incomplete because of https://github.com/getsentry/sentry-docs/pull/6617/commits/c0e507a7c70dc4c54540b265679b816c023abfc7#r1193011580.

File: src/platforms/python/common/configuration/integrations/loguru.mdx (hunk 1 above)
Comment (PerchunPak) [reply src/platforms/python/common/configuration/integrations/loguru.mdx]: Fixed in 63f4e09e46f0ad7eb6fa2f46769f376780289f68. Thank you!

File: src/platforms/python/common/configuration/integrations/loguru.mdx (hunk 3 above)
Comment (PerchunPak) [reply src/platforms/python/common/configuration/integrations/loguru.mdx:19]: Fixed in 63f4e09e46f0ad7eb6fa2f46769f376780289f68. Thank you!

File: src/platforms/python/common/configuration/integrations/loguru.mdx (hunk 4 above)
Comment (PerchunPak) [reply src/platforms/python/common/configuration/integrations/loguru.mdx]: Fixed in 63f4e09e46f0ad7eb6fa2f46769f376780289f68. Thank you!

File: src/platforms/python/common/configuration/integrations/loguru.mdx (hunk 5 above)
Comment (PerchunPak) [reply src/platforms/python/common/configuration/integrations/loguru.mdx]: Fixed in 63f4e09e46f0ad7eb6fa2f46769f376780289f68. Thank you!

File: src/platforms/python/common/configuration/integrations/loguru.mdx (hunk 6 above)
Comment (PerchunPak) [reply src/platforms/python/common/configuration/integrations/loguru.mdx]: Fixed in 63f4e09e46f0ad7eb6fa2f46769f376780289f68. Thank you!

File: src/platforms/python/common/configuration/integrations/loguru.mdx (hunk 7 above)
Comment (PerchunPak) [reply src/platforms/python/common/configuration/integrations/loguru.mdx]: Fixed in 63f4e09e46f0ad7eb6fa2f46769f376780289f68. Thank you!

File: src/platforms/python/common/configuration/integrations/loguru.mdx (hunk 9 above)
Comment (PerchunPak) [reply src/platforms/python/common/configuration/integrations/loguru.mdx]: Fixed in 63f4e09e46f0ad7eb6fa2f46769f376780289f68. Thank you!

**Comment by PerchunPak** [comment]
Sorry for such delay, the life happened.

File: src/platforms/python/common/configuration/integrations/loguru.mdx (hunk 10 above)
Comment (antonpirker) [reply src/platforms/python/common/configuration/integrations/loguru.mdx]: I checked, they are in the final version.

**Comment by antonpirker** [comment]
Hey @PerchunPak ! No worries, life is happening overhere also all the time :-)

**Comment by antonpirker** [comment]
This is now ready, and I will merge when the Loguru integration is released!

**Comment by PerchunPak** [comment]
I will look at changes in the evening too
